gae/room/protorpc_messages.py

```diff
@@ -991,6 +991,56 @@ class SiteCaptain(messages.Message):
   type = messages.StringField(4)
 
 
+############
+# StaffPosition #
+############
+def StaffPositionModelToMessage(mdl):
+  s = StaffPosition(
+      id=mdl.key.integer_id(),
+      name=mdl.position_name,
+    )
+  if mdl.hourly_rate_after_date:
+    s.hourly_rates = [{'date': str(dr[0]), 'rate': float(dr[1])} for dr in (
+          dr.split() for dr in mdl.hourly_rate_after_date if dr)]
+  else:
+    s.hourly_rates = [{'date': "", 'rate': mdl.hourly_rate}]
+  if mdl.mileage_rate_after_date:
+    s.mileage_rates = [{'date': str(dr[0]), 'rate': float(dr[1])} for dr in (
+                dr.split() for dr in mdl.mileage_rate_after_date if dr)]
+  else:
+    s.mileage_rates = [{'date': "", 'rate': mdl.mileage_rate}]
+  return s
+
+
+def StaffPositionMessageToModel(msg, mdl):
+  if not msg.name:
+    raise remote.ApplicationError('position name is required')
+  mdl.position_name = msg.name
+
+  mdl.hourly_rate_after_date = [h.date + " " + str(h.rate) for h in msg.hourly_rates if h.rate and h.date]
+  mdl.hourly_rate = mdl.GetHourlyRate(datetime.datetime.today())
+
+  mdl.mileage_rate_after_date = [m.date + " " + str(m.rate) for m in msg.mileage_rates if m.rate and m.date]
+  mdl.mileage_rate = mdl.GetMileageRate(datetime.datetime.today())
+
+  return mdl
+
+
+class RateAfterDate(messages.Message):
+  date = messages.StringField(1)
+  rate = messages.FloatField(2)
+
+class StaffPosition(messages.Message):
+  id = messages.IntegerField(1)
+  name = messages.StringField(2)
+  hourly_rate = messages.FloatField(3)
+  hourly_date = messages.StringField(4)
+  mileage_rate = messages.FloatField(5)
+  mileage_date = messages.StringField(6)
```

Comment:
Can we simplify even further and eliminate hourly_rate, hourly_date, mileage_rate_, mileage_date ?  and just have hourly_rates and mileage_rates?

Reply:
Absolutely!

Reply:
This is where I have been stuck because I'm really finagling the hourly_rates/ mileage_rates to work with the basic cru_api.  It just seems like I'm missing something. Should I use another api?  choices_api, maybe?